app/src/app/[lng]/admin/organization/[id]/projects/CreateEditProjectModal.tsx

```diff
@@ -0,0 +1,264 @@
+import FormattedNumberInput from "@/components/formatted-number-input";
+import { MdWarning } from "react-icons/md";
+import { api } from "@/services/api";
+import { UseErrorToast, UseSuccessToast } from "@/hooks/Toasts";
+
+interface CreateEditProjectModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  t: TFunction;
+  onOpenChange: (val: boolean) => void;
+  organizationId: string;
+  projectData?: {
+    projectId: string;
+    projectName: string;
+    description: string;
+    cityCountLimit: number;
+  } | null;
+}
+
+const schema = z.object({
+  projectName: z.string().min(3, "required"),
+  description: z.string().min(3, "required"),
+  cityCountLimit: z.number().min(1, "required"),
+});
+
+type Schema = z.infer<typeof schema>;
+
+const CreateEditProjectModal: FC<CreateEditProjectModalProps> = ({
+  isOpen,
+  onClose,
+  t,
+  onOpenChange,
+  projectData,
+  organizationId,
+}) => {
+  const [step, setStep] = useState(1);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    watch,
+    setError,
+    clearErrors,
+    setFocus,
+    setValue,
+    control,
+    getValues,
+    trigger,
+    formState: { errors, isValid, isDirty },
+  } = useForm<Schema>({
+    mode: "all",
+    resolver: zodResolver(schema),
+  });
+
+  const { showErrorToast } = UseErrorToast({
+    title: t("error-message"),
+  });
+  const { showSuccessToast } = UseSuccessToast({
+    title: projectData ? t("project-updated") : t("project-created"),
+    duration: 1200,
+  });
+
+  useEffect(() => {
+    if (projectData) {
+      reset({
+        projectName: projectData.projectName,
+        description: projectData.description,
+        cityCountLimit: Number(projectData.cityCountLimit),
+      });
+    } else {
+      reset({
+        projectName: "",
+        description: "",
+        cityCountLimit: 0,
+      });
+    }
+  }, [reset, projectData]);
+
+  const [createProject, { isLoading: isProjectLoading }] =
+    api.useCreateProjectMutation();
+
+  const [editProject, { isLoading: isEditLoading }] =
+    api.useEditProjectMutation();
+
+  const isSubmitting = isProjectLoading || isEditLoading;
+
+  const closeFunction = () => {
+    onClose();
+    reset();
+  };
+
+  const handleFormSubmit = async (data: Schema) => {
+    const { projectName, description, cityCountLimit } = data;
+
+    let response = null;
+    if (projectData) {
+      response = await editProject({
+        projectId: projectData.projectId,
+        name: projectName,
+        description,
+        cityCountLimit,
+      });
+    } else {
+      response = await createProject({
+        name: projectName,
+        description,
+        cityCountLimit,
+        organizationId,
+      });
+    }
+
+    if (response.data) {
+      showSuccessToast();
+      closeFunction();
+    } else {
+      showErrorToast();
+    }
+  };
+
+  return (
+    <DialogRoot
+      preventScroll
+      open={isOpen}
+      onOpenChange={(e: any) => {
```

Comment:
You can probably leave out the `: any` here since the component's type definition allows the actual type to be inferred